File: Recursion/basic.java
class Test {
    
    static void func1() {
        System.out.println("func1");
    } 

    static void func2() {
        System.out.println(" Before func1");
        func1();
        System.out.println(" After func1");
    }
    public static void main(String[] args) {
        
        System.out.println("Before func2");
        func2();
        System.out.println("After func2");
    }
}

File: Recursion/basic.cpp
#include<iostream>
using namespace std;

void func1() {
    cout << "func1" << endl;
}

void func2() {
    cout << "Before Func 1" << endl;
    func1();
    cout << "After Func 1" << endl;
}

int main() {
    cout << "Before Func 2" << endl;
    func2();
    cout << "After Func 2" << endl;
}